### DISCUSSION TITLE
Export PaginatedDocs

### DISCUSSION THREAD

**Discussion by edtorba** [payloadcms/payload#9547] [Feature Requests & Ideas]
Hi everyone!

I wanted to start a discussion about the possibility of including the `PaginatedDocs` type in the generated `payload-types.ts` file. Since many responses from Payload CMS collections are paginated, having a generic `PaginatedDocs<T>` type automatically included could streamline development and ensure consistency. This type, which includes properties like docs, totalDocs, limit, totalPages, page, etc., is widely applicable and would save developers from manually defining it or adding it to separate utility files.

This becomes especially practical in setups where the Payload CMS (head) is separate from the client-facing site, and the generated payload-types.ts file is part of an independent package (e.g., @repo/head-types). Including PaginatedDocs in the generated models would make it easier to share a complete and reusable type system across projects without additional manual work.

Would it make sense to incorporate PaginatedDocs as a standard part of the type generation process? Looking forward to hearing thoughts from the team and the community!

**Comment by r1tsuu** [comment]
You can perfectly fine install the `payload` package as a dev dependency and use types from it, including `PaginatedDocs`.

**Reply by edtorba** [reply]
Thanks for the suggestion! While I understand that `PaginatedDocs` is available directly from the `payload` package, using `payload` as a dev dependency introduces a version coupling between the backend and the client-facing or shared type package e.g., `@repo/head-types`. This requires the `payload` package version in my independent type package to always remain perfectly in sync with the backend version.

One of the goals of generating `payload-types.ts` into an independent package is to decouple the client-facing codebase from Payload CMS’s version changes. Including `PaginatedDocs` directly in the generated types would make the shared package more self-contained and eliminate the need to add `payload` as a dev dependency altogether, maintaining a clean separation of concerns.
